webapp-django/crashstats/crashstats/migrations/0003_1457747_graphics_data_migration.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.15 on 2018-09-25 15:42
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from django.db import connection, migrations
+
+
+def copy_graphics_devices_data(apps, schema_editor):
+    """Copy data from the old sqlalchemy/alembic graphics_device table to the Django one"""
+    GraphicsDevice = apps.get_model('crashstats', 'GraphicsDevice')
+
+    cursor = connection.cursor()
+
+    # First, we verify the table is there, if not, we don't need to do anything
+    cursor.execute("""
+    SELECT relname from pg_catalog.pg_class
+    WHERE relname = 'graphics_device'
+    """)
+    row = cursor.fetchone()
+    if row is None:
+        print('no graphics_device table--nothing to do', end='')
+        return
+
+    # Second, pull all the data from it
+    columns = ['vendor_hex', 'adapter_hex', 'vendor_name', 'adapter_name']
+    cursor.execute(
+        """
+        SELECT %(columns)s
+        FROM graphics_device
+        """ % {
+            'columns': ', '.join(columns)
+        }
+    )
+
+    print('')
+    insert_count = 0
+    skip_count = 0
+
+    # Third, create new GraphicsDevice instances and save them to the db.
+    # Looks like stage and prod have a *lot* of junk data. This is a lookup
+    # table so we're going to junk anything that doesn't have names.
+    for row in cursor.fetchall():
+        item = dict(zip(columns, row))
+        if item['vendor_name'] is None and item['adapter_name'] is None:
```

Comment:
I figure if both of these columns are nulls, then this row isn't doing anything helpful and we should ditch it.